packages/at_secondary_server/lib/src/enroll/enrollment_manager.dart

```diff
@@ -4,26 +4,28 @@ import 'package:at_commons/at_commons.dart';
 import 'package:at_persistence_secondary_server/at_persistence_secondary_server.dart';
 import 'package:at_secondary/src/constants/enroll_constants.dart';
 import 'package:at_secondary/src/enroll/enroll_datastore_value.dart';
-import 'package:at_secondary/src/server/at_secondary_impl.dart';
 import 'package:at_secondary/src/utils/secondary_util.dart';
 import 'package:at_utils/at_logger.dart';
+import 'package:meta/meta.dart';
 
 /// Manages enrollment data in the secondary server.
 ///
 /// This class provides methods to retrieve and store enrollment data
 /// associated with a given enrollment ID. It interacts with the
 /// SecondaryKeyStore to persist and retrieve enrollment information.
 class EnrollmentManager {
-  final SecondaryKeyStore _keyStore;
+  final SecondaryKeyStore<String, AtData?, AtMetaData?> keyStore;
+  final String atSign;
 
-  final logger = AtSignLogger('AtSecondaryServer');
+  /// log messages here are important, so hard-coding level to 'info'
+  final AtSignLogger logger = AtSignLogger('EnrollmentManager')..level = 'info';
 
   /// Creates an instance of [EnrollmentManager].
   ///
   /// The [keyStore] is required to interact with the persistence layer.
-  EnrollmentManager(this._keyStore);
+  EnrollmentManager(this.keyStore, this.atSign);
```

Comment:
The primary changes to implement the per-enrollment-data feature are here in EnrollmentManager.